src/Raven.Server/Documents/IndexWaiter.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Sparrow.Server;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents;
+
+public class IndexWaiter : IDisposable
```

Comment:
can you call it ClusterIndexWaiter/RaftIndexWaiter or sth like that? Would like to avoid mixing this with Indexes

Reply:
The implementation is very generic and theoretically can be used for any incremental only long-value 
But I will change it to `RaftIndexWaiter`

Reply:
done